types/ws/index.d.ts

```diff
@@ -12,134 +12,171 @@
 
 /// <reference types="node" />
 
-import * as events from 'events';
-import * as http from 'http';
-import * as https from 'https';
-import * as net from 'net';
-import * as url from 'url';
-import * as zlib from 'zlib';
-import * as stream from 'stream';
-import { SecureContextOptions } from 'tls';
+import { EventEmitter } from "events";
+import {
+    Agent,
+    ClientRequest,
+    ClientRequestArgs,
+    IncomingMessage,
+    OutgoingHttpHeaders,
+    Server as HTTPServer,
+} from "http";
+import { Server as HTTPSServer } from "https";
+import { Socket } from "net";
+import { Duplex, DuplexOptions } from "stream";
+import { SecureContextOptions } from "tls";
+import { URL } from "url";
+import { ZlibOptions } from "zlib";
 
 // WebSocket socket.
-declare class WebSocket extends events.EventEmitter {
+declare class WebSocket extends EventEmitter {
     /** The connection is not yet open. */
-    static CONNECTING: 0;
+    static readonly CONNECTING: 0;
     /** The connection is open and ready to communicate. */
-    static OPEN: 1;
+    static readonly OPEN: 1;
     /** The connection is in the process of closing. */
-    static CLOSING: 2;
+    static readonly CLOSING: 2;
     /** The connection is closed. */
-    static CLOSED: 3;
+    static readonly CLOSED: 3;
 
-    binaryType: string;
-    bufferedAmount: number;
-    extensions: string;
-    protocol: string;
+    binaryType: "nodebuffer" | "arraybuffer" | "fragments";
```

Comment:
Please see here for declared binary types https://github.com/websockets/ws/blob/fc7e27d12ad0af90ce05302afc85c292024000b4/lib/constants.js#L4